authors/apps/favorites/views.py

```diff
@@ -0,0 +1,88 @@
+from rest_framework import status
+from rest_framework.views import APIView
+from .serializers import FavoritesSerializer, FavoritesArticleSerializer
+from rest_framework.response import Response
+from authors.apps.articles.models import Article
+from authors.apps.articles.serializers import ArticleSerializer
+from rest_framework.permissions import IsAuthenticated
+from .models import Favorites
+
+
+class favorite(APIView):
+    """
+    Function to handle favorite articles functionality
+    """
+    permission_classes = (IsAuthenticated,)
+
+    def get(self, request, article_id):
+        # toggle aticle as favorite or not favorite
+
+        article = Article.objects.filter(pk=article_id).distinct()
+        if not article.exists():
+            return Response(
+                {
+                    'error': 'Article not found.'
+                },
+                status=status.HTTP_400_BAD_REQUEST,)
+
+        favorite = Favorites.objects.filter(article=article_id, user=request.user.id)
+
+        article = article.first()
+        favs_count = None
+        message = None
+
+        if favorite.exists():
+            # fav record exists
+
+            favorite.first().delete()
+            favs_count = article.favorite_count - 1
+            message = "Article removed from favorites."
+
+        elif not article.author.id == request.user.id:
+            # Check that request is not from the article author
+
+            serializer = FavoritesSerializer(data={
+                'article':article_id,
+                'user':request.user.id})
+            if serializer.is_valid():
+                serializer.save()
+
+            favs_count = article.favorite_count + 1
+            message = "Article added to favorites."
+        else:
+
+            # this runs when request is from article owner
+            message = "You can not favorite your own article!"
+            return Response({'error': message},
+                            status=status.HTTP_400_BAD_REQUEST,)
+
+        article_serializer = FavoritesArticleSerializer(
+            article, data={'favorite_count': favs_count}, partial=True
+        )
+        article_serializer.is_valid(raise_exception=True)
+        article_serializer.save()
+
+        return Response({'message': message},
+                        status=status.HTTP_200_OK,)
+
+
+class favorites(APIView):
+    """
+    Return favorite articles
+    """
+    permission_classes = (IsAuthenticated,)
+
+    def get(self, request):
+
+        favs = Favorites.objects.filter(user=request.user)
```

Comment:
@joelmugaya look into `get_queryset` or `get_object` method so we can have separation of concerns.